app/controllers/api/v1/books_controller.rb

```diff
@@ -15,6 +15,8 @@ def show
         book = Book.find(params[:id])
 
         render json: book
+      rescue ActiveRecord::RecordNotFound
```

Comment:
Don't rescue in the controller method, use `rescue_from` instead:

https://apidock.com/rails/ActiveSupport/Rescuable/ClassMethods/rescue_from

This helps to avoid code repetition and also clean up your controllers a bit.